### ISSUE TITLE
Add a multitenant auth options sample

### ISSUE THREAD

**Issue by HaoK** [aspnet/AuthSamples#33]
See https://github.com/aspnet/Security/issues/1718 for more context

**Comment by aagarwal10** [comment]
hi @HaoK - I have scenario of multi-tenant auth, where I need to dynamically add OAuth2 Scheme based on tenant (each tenant has its own OAuth Options) and wire it with cookie scheme for my aspnet core web app (not implemented with any Javascript framework). Please suggest if this is possible with these auth samples. Thanks

**Comment by HaoK** [comment]
Try doing something like this sample https://github.com/aspnet/AuthSamples/pull/44

**Comment by aagarwal10** [comment]
@HaoK - Is it also possible to add Auth Handler dynamically, similar to adding/removing auth scheme from a controller?

for example in a controller action method, can we do something like?

.AddScheme<SimpleOptions, SimpleAuthHandler>("fromAction", o => o.DisplayMessage = "I am from controller action.") 

Also, even if this possible, how can this whole dynamic auth handler registration via action method be done for application running in a web farm. 
--------

Actually I have a case where a tenant gets dynamically registered to the web application passing there OAuth Options on the fly. This is scenario of SMART on FHIR (HL7 Standard)

**Comment by HaoK** [comment]
Yes, you can just use the IAuthenticationSchemeProvider and call AddScheme, but you would probably have to have your implementation of this use some kind of shared data store backing the implementation to make them consistent across the web farm.

**Comment by muratg** [comment]
@HaoK will this be in 2.2, or should we move this out?

**Comment by HaoK** [comment]
You can move it out, there's some weird build flakiness that this sample introduces, I will merge it into master once its consistently passing on CI
